libs/cardano-ledger-core/src/Cardano/Ledger/DRep.hs

```diff
@@ -142,7 +148,22 @@ instance Crypto c => EncCBOR (DRepState c) where
         !> To drepAnchor
         !> To drepDeposit
 
-instance Crypto c => ToJSON (DRepState c)
+instance Crypto c => ToJSON (DRepState c) where
+  toJSON x@(DRepState _ _ _) =
+    let DRepState {..} = x
+     in toJSON $
+          KeyMap.fromList
+            [ "expiry" .= toJSON drepExpiry
+            , "anchor" .= toJSON drepAnchor
+            , "deposit" .= toJSON drepDeposit
+            ]
```

Comment:
Avoid including optional anchor when it is not present and using `object` is the proper way way of constructing objects in Aeson:
```suggestion
          object $
            [ "expiry" .= toJSON drepExpiry
            , "deposit" .= toJSON drepDeposit
            ] ++ ["anchor" .= toJSON anchor | SJust anchor <- [drepAnchor]]
```